packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1038,3 +1039,81 @@ export function getClientNameOverride(
 ): string | undefined {
   return getScopedDecoratorData(context, clientNameKey, entity, languageScope);
 }
+
+const overrideClientMethodKey = createStateSymbol("overrideClientMethod");
+
+// Recursive function to collect parameter names
+function collectParams(properties: RekeyableMap<string, ModelProperty>): ModelProperty[] {
+  let params: ModelProperty[] = [];
+
+  properties.forEach((value, key) => {
+    // If the property is of type 'model', recurse into its properties
+    if (value.type.kind === "Model") {
+      params = params.concat(collectParams(value.type.properties));
```

Comment:
will this cause infinite loop if cyclic ref?

Reply:
good catch, let me make sure we're not duplicating these properties